exist-core/src/main/java/org/exist/xquery/functions/util/LogFunction.java

```diff
@@ -147,13 +146,21 @@ public Sequence eval(Sequence[] args, Sequence contextSequence) throws XPathExce
         // Iterate over all input
         while (i.hasNext()) {
             final Item next = i.nextItem();
-            try (StringWriter writer = new StringWriter()) {
-                XQuerySerializer xqs = new XQuerySerializer(context.getBroker(), props, writer);
-                xqs.serialize(next.toSequence());
-                buf.append(writer.toString());
 
-            } catch (IOException | SAXException e) {
-                throw new XPathException(this, e.getMessage());
+            if(next instanceof StringValue){
+                // Use simple string value
+                buf.append(next.getStringValue());
+
+            } else {
+                // Serialize data
+                try (StringWriter writer = new StringWriter()) {
+                    XQuerySerializer xqs = new XQuerySerializer(context.getBroker(), props, writer);
+                    xqs.serialize(next.toSequence());
+                    buf.append(writer.toString());
+
+                } catch (IOException | SAXException e) {
+                    throw new XPathException(this, e.getMessage());
+                }
             }
 
         }
```

Comment:
Looks like this file was added to the PR by mistake due to your branch not being in sync with upstream? The change to util:log has already been merged.

Reply:
sorry try to fix that

Reply:
As soon as that is fixed, I am happy to merge this.